cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -347,28 +347,28 @@ public ApplicationWithPrograms deployApp(Id.Namespace namespace, @Nullable Strin
    * @param namespaceId the {@link Id.Namespace} under which all application should be deleted
    * @throws Exception
```

Comment:
hum... after looking at the call sequence, why not having this method just throw a `CannotBeDeletedException` with the error message containing the joined string of all the program ids that are still live (basically the logic in the `ProgramLifecycleHandler`).